### PR TITLE
browser: Always restore session by default.

### PR DESCRIPTION
**Pull request by Ambrevar**
@aartaka Does that make sense for you?

@jmercouris @aadcg Tests welcome! :)

### CODE REVIEW

File: source/renderer/gtk.lisp [hunk 1]
```diff
@@ -163,11 +163,11 @@ As a workaround, we never leave the GTK main loop when running from a REPL.
 
 See https://github.com/atlas-engineer/nyxt/issues/740")
 
-(defun renderer-thread-p ()
+(defun renderer-thread-p (&optional (thread (bt:current-thread)))
   #+darwin
   (string= "main thread" (bt:thread-name (bt:current-thread)))
```
Comment (jmercouris) [inline source/renderer/gtk.lisp:168]: Why not also use `thread` here instead of `bt:current-thread`?

**Comment by jmercouris** [comment]
Also, I like the previous behavior, I see this as a regression in functionality! I suggest we do not merge this pull request and just continue with 3-pre-release-1

**Comment by aartaka** [comment]
@jmercouris, I guess we need to understand context of this change: 
- The existing startup sequence is overly complicated, not least because of the history prompting.
- Most people disable prompting in favor of either always restoring session (Pierre) 
  - or never restoring it (myself).
- And that history restoration is always available as a command (`restore-session-by-name`, currently broken, to-be-fixed).

It's a reasonable step then to remove the unnecessary, duplicative, and bothersome prompting from the startup.

**Comment by Ambrevar** [comment]
@jmercouris I'll remove this functionality for now, because we can add after 3.0 release, but the other way around we require 4.0.

We need more time to polish our startup sequence and for the time being, history prompting so many edge cases that it's really prone to prevent Nyxt from starting at all.

**Comment by aadcg** [comment]
I agree that the session should be restored by default. 

What I don't understand is under which circumstances will be the `new` buffer be shown? I deleted all buffers and restarted Nyxt, but I still couldn't get it. Am I missing something?

**Comment by Ambrevar** [comment]
The `default-new-buffer-url` is shown if there is nothing to restore and if you do not pass URLs to the  command line on startup.  @aadcg If it does not behave this way for you, could you share a recipe?

**Comment by Ambrevar** [comment]
@jmercouris Also note that the "prompt on startup" feature has been broken for months and no one has ever complained xD

**Comment by aadcg** [comment]
> If it does not behave this way for you, could you share a recipe?

I manually deleted `~/.local/share/nyxt/history`.  Then started Nyxt from the REPL, deleted all buffers, quitted, and restarted from the REPL.

**Comment by Ambrevar** [comment]
> Also, `history/default.lisp` seems to be re-populated after deletion. Where is all of this data coming from?

You must delete it after quitting Nyxt ;)

**Comment by aadcg** [comment]
Ok, all good!

**Comment by aartaka** [comment]
> @aartaka This pull request should fix history saving and restoration, doesn't it?


Aaaaaah, I see 84b9775f83507cb8c9afc10778effeee6c928f65!

**Comment by aadcg** [comment]
@Ambrevar working as expected! Even before these last 3 commits, it was working fine. It was me, who didn't get rid of the history file properly.

File: source/manual.lisp [hunk 2]
```diff
@@ -552,7 +552,7 @@ exec nyxt --headless --no-auto-config --profile nosave --config \"$0\"
 
 ;; Nyxt shows a history restore prompt by default, you don't want it to.
```
Comment (aartaka) [inline source/manual.lisp:553]: Fix the comment too :P

File: source/browser.lisp [hunk 3]
```diff
@@ -321,10 +310,15 @@ restored."
     ;; `web-buffer's.
     (mapcar #'window-delete (window-list))
     (window-make browser)
-    ;; Restore session before opening command line URLs, otherwise it will
-    ;; reset the session with the new URLs.
-    (restore-session)
-    (load-start-urls urls)
+    (if (restore-session-on-startup-p *browser*)
```
[the remark this answers is not in the document]
Comment (Ambrevar) [reply source/browser.lisp:313]: Does not work here unless we store the result of `restore-history-buffers` in an intermediate variable.

File: source/browser.lisp [hunk 4]
```diff
@@ -360,7 +354,7 @@ This is useful to tell REPL instances from binary ones."
                      url))))
 
 ;; REVIEW: Do we need :NO-FOCUS? It's not used anywhere.
-(-> open-urls ((cons quri:uri *) &key (:no-focus boolean)) *)
+(-> open-urls ((maybe (cons quri:uri *)) &key (:no-focus boolean)) *)
```
[the remark this answers is not in the document]
Comment (Ambrevar) [reply source/browser.lisp:357]: Maybe we should use this library instead which also offers `list-of`: https://gitlab.common-lisp.net/asdf/asdf-finalizers.

**Comment by Ambrevar** [comment]
Merged with 146817527b11f0bf13a7278e2a3a86600dcfb525.

**Comment by jmercouris** [comment]
> @jmercouris Also note that the "prompt on startup" feature has been broken for months and no one has ever complained xD

I would have complained, but I am not on master :-P

**Pull request closed without merging**